pkg/sqlqueries/SQL/mto/milestone_solution_link/merge_solutions_to_milestone.sql

```diff
@@ -0,0 +1,41 @@
+WITH source_links AS (
+    SELECT 
+        CAST(:milestone_id AS UUID) AS milestone_id,
+        UNNEST(CAST(:solution_ids AS UUID[])) AS solution_id,
+        CAST(:created_by AS UUID) AS created_by
+),
+
+inserted_links AS ( --noqa
+    INSERT INTO mto_milestone_solution_link (milestone_id, solution_id, created_by)
+    SELECT milestone_id, solution_id, created_by
+    FROM source_links
+    ON CONFLICT (milestone_id, solution_id) DO NOTHING
+    RETURNING solution_id
+),
+
+deleted AS ( --noqa
+    DELETE FROM mto_milestone_solution_link
+    WHERE
+        milestone_id = :milestone_id
+        AND solution_id NOT IN (SELECT solution_id FROM source_links)
+    RETURNING solution_id
+)
+
+SELECT
+    s.id,
+    s.model_plan_id,
+    s.mto_common_solution_key,
+    s.name,
```

Comment:
I noticed that this should join the common solution table, I'll push up a quick fix to ensure we get proper data.

Reply:
updated!